pkg/cloudprovider/providers/aws/aws_routes.go

```diff
@@ -56,17 +62,33 @@ func (s *AWSCloud) ListRoutes(clusterName string) ([]*cloudprovider.Route, error
 	}
 
 	var routes []*cloudprovider.Route
+	var instanceIDs []*string
+
 	for _, r := range table.Routes {
-		instanceID := orEmpty(r.InstanceId)
-		destinationCIDR := orEmpty(r.DestinationCidrBlock)
+		instanceID, destinationCIDR := getInstanceAndCIDR(r)
 
 		if instanceID == "" || destinationCIDR == "" {
 			continue
 		}
 
-		instance, err := s.getInstanceById(instanceID)
-		if err != nil {
-			return nil, err
+		instanceIDs = append(instanceIDs, &instanceID)
+	}
+
+	instances, err := s.getInstancesByIDs(instanceIDs)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range table.Routes {
+		instanceID, destinationCIDR := getInstanceAndCIDR(r)
```

Comment:
Ah - I guess you introduced the function because it is used in two places here.  Still not sure it's worth it (particularly as we don't really use destinationCIDR above), but I do understand now!